ui/apps/platform/src/Containers/Network/Header/NetworkSearch.js

```diff
@@ -1,29 +1,69 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { connect } from 'react-redux';
 import { createStructuredSelector } from 'reselect';
 
 import { selectors } from 'reducers';
 import { actions as pageActions } from 'reducers/network/page';
 import { actions as searchActions } from 'reducers/network/search';
+import useURLSearch from 'hooks/useURLSearch';
+import searchOptionsToQuery from 'services/searchOptionsToQuery';
+import { getSearchOptionsForCategory } from 'services/SearchService';
+import { isCompleteSearchFilter } from 'utils/searchUtils';
+import SearchFilterInput from 'Components/SearchFilterInput';
 import {
     ORCHESTRATOR_COMPONENT_KEY,
     orchestratorComponentOption,
 } from 'Containers/Navigation/OrchestratorComponentsToggle';
-import ReduxSearchInput from 'Containers/Search/ReduxSearchInput';
 
 import './NetworkSearch.css';
 
+function searchFilterToSearchEntries(searchFilter) {
+    const entries = [];
+    Object.entries(searchFilter).forEach(([key, value]) => {
+        entries.push({ label: `${key}:`, value: `${key}:`, type: 'categoryOption' });
+        if (value !== '') {
+            const values = Array.isArray(value) ? value : [value];
+            const valueOptions = values.map((v) => ({ label: v, value: v }));
+            entries.push(...valueOptions);
+        }
+    });
+    return entries;
+}
+
+const searchCategory = 'DEPLOYMENTS';
+const searchOptionExclusions = ['Cluster', 'Namespace', 'Namespace ID', 'Orchestrator Component'];
+
 function NetworkSearch({
-    searchOptions,
-    searchModifiers,
     selectedNamespaceFilters,
-    setSearchOptions,
-    setSearchSuggestions,
+    dispatchSearchFilter,
     closeSidePanel,
     isDisabled,
 }) {
+    const [searchOptions, setSearchOptions] = useState([]);
+    const { searchFilter, setSearchFilter } = useURLSearch();
+
+    useEffect(() => {
+        const { request, cancel } = getSearchOptionsForCategory(searchCategory);
+        request
+            .then((options) => {
+                const filteredOptions = options.filter((o) => !searchOptionExclusions.includes(o));
+                setSearchOptions(filteredOptions);
+            })
+            .catch(() => {
+                // A request error will disable the search filter.
+            });
+
+        return cancel;
+    }, [setSearchOptions]);
+
+    // Keep the Redux store in sync with the URL Search Filter
```

Comment:
@dvail could you refresh my memory about the plan for search in this section in terms of URL and redux state? I think someone (maybe you) mentioned that we'd want to keep the URL saved in redux because we'd have to filter again when we switch to a different page and come back. I was also thinking that we have cases where you can navigate to the Network Graph page from other parts of the UI and when that navigation happens, sometimes we also modify the URL to filter by some values, etc. Do you think the syncing might become a bit messy in those cases and maybe cause weird edge cases? I guess I'm worried about the hassle of keeping things in sync vs. having one source of truth. But if that's manageable, and easy, then no worries

Reply:
We talked about this a little bit on our UX sync this past Monday and decided it would probably be best to do either Redux _or_ URL state - mixing the two probably leads to too many edge cases that would cause bugs or confusing behavior for the user.

I believe the tentative recommendation from @alanonthegit was that URL parameters is probably the direction we want to head, especially with the larger upcoming changes to the network graph. I'm going to keep working on this PR to see what implementation of the cluster and namespace filters in the URL looks like, both from a UX perspective and a code perspective.

We figured the pros and cons would boil down to something like this:

### State stored in Redux

**Pros**
- Allows the user to retain their place in the Network Graph regardless of how and when they navigate to other pages in the app

**Cons**
- Makes links or other actions where we want to set a series of parameters more difficult to reconcile with any in-memory state
- Inconsistent with the UX of how we do this in other place of the app

### URL Parameters

**Pros**
- Allows the user to navigate to other pages and _back_ to the net graph to retain state
- Allows the user to easily "undo" filter changes with the back button
- Allows easier linking to the net graph with a canonical URL to drive the state
- Makes sharing the state of a net graph between users easy

**Cons**
- Users that navigate away and then back _without_ using the browser's history lose all state
- This would change a piece of functionality users might be accustomed to

Note that the current functionality is sort of an incomplete hybrid between the two. The selected cluster and namespaces persist in memory, but search filters are lost on navigation. Meanwhile you can also set search filters via the URL (or the passthrough global search), but changes past the initial page load do not stick or update the URL.

I'm definitely in favor of the single source of truth, and leaning toward URL params as the better approach, but it depends on whether or not the loss of the existing in-memory functionality is acceptable at this point.